### PR TITLE
Fix host case comparison

### LINKED ISSUE #1600: Could not load file or assembly 'dotnet.EXE.dll'
I downloaded net cli binaries and executed dotnet restore within folder which contains asp.net core project. Some packages were restored and several errors about incompativlity were displayed.
Then I tried execute dotnet new within empty directory and got error
`Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'dotnet.EXE.dll' or one of its dependencies.`
I tried to uninstall and reinstall (and also repair) but error still occurs.

version 1.0.0.001540

### PR DESCRIPTION
**Pull request by schellap**
Fixes #1600

### CODE REVIEW

**Comment by davidfowl** [comment]
Add a test?

**Comment by schellap** [comment]
Filed an issue for RTM to get back to it.

**Comment by schellap** [comment]
@piotrpMSFT or @gkhanna79 PTAL.

**Comment by davidfowl** [comment]
Why not add a test now?

**Comment by schellap** [comment]
Right now, I don't want to spend time learning how cli test infrastructure works as there are other higher pri bugs/work items on my plate. If an accompanying test helps completeness, I will close this PR and come back to it later, but having a tracking bug does the same...

File: src/corehost/common/pal.h [hunk 1]
```diff
@@ -95,6 +95,9 @@ namespace pal
 
     inline int strcmp(const char_t* str1, const char_t* str2) { return ::wcscmp(str1, str2); }
     inline int strcasecmp(const char_t* str1, const char_t* str2) { return ::_wcsicmp(str1, str2); }
+    inline int strncmp(const char_t* str1, const char_t* str2, int len) { return ::wcsncmp(str1, str2, len); }
+    inline int strncasecmp(const char_t* str1, const char_t* str2, int len) { return ::_wcsnicmp(str1, str2, len); }
```
Comment (gkhanna79) [inline src/corehost/common/pal.h]: We are doing case-insensitive comparison here but not in the !Win32 case?

**Comment by gkhanna79** [comment]
LGTM other than one comment.

File: src/corehost/common/pal.h (hunk 1 above)
Comment (schellap) [inline src/corehost/common/pal.h]: `strcasecmp` is a funny name but used by unix, Ruby as `casecmp` and Python as `casefold`

**Comment by schellap** [comment]
@davidfowl, merging... but I will definitely get to the issue #1682

**Pull request merged**